### PR TITLE
Add a minimum ability modifier to familiars

### PR DESCRIPTION
**Pull request by Duncan-Henry**
Per issue #14776, familiars should have a minimum of 3 + master level in perception, acrobatics, and stealth. This change is a first draft at fixing that.

How are localizations handled on the familiar sheet? Is there a way to generate a "Minimum Ability Modifier" label via localizations rather than the english i've hardcoded in there?

### CODE REVIEW

**Comment by CarlosFdez** [comment]
If you're gonna do a hotswap of modifier, it will be better to do `const attributeModifier = masterAttributeModifier > 2 ? new ModifierPF2e({ params }) : new ModifierPF2e({ params })` and then pass that around. Modifier objects aren't breakdowns, breakdowns are text summaries.

For the localization, they're in `en.json`. Somewhere in PF2E.Actor.Familiar should suffice.

And finally, use `npm run lint:fix` before pushing it up. It'll correct your formatting mistakes (and let it pass through the workflow once I enable it)

**Comment by Duncan-Henry** [comment]
> If you're gonna do a hotswap of modifier, it will be better to do `const attributeModifier = masterAttributeModifier > 2 ? new ModifierPF2e({ params }) : new ModifierPF2e({ params })` early on and then pass that around. Modifier objects aren't breakdowns, breakdowns are text summaries.
> 
> For the localization, they're in `en.json`. Somewhere in PF2E.Actor.Familiar should suffice.
> 
> And finally, use `npm run lint:fix` before pushing it up. It'll correct your formatting mistakes (and let it pass through the workflow once I enable it)

Thanks for the feedback! Changes have been made, let me know if I missed anything.

**Comment by CarlosFdez** [comment]
Looks solid, though we can't merge it until v12, and we might need to support pets before we can actually merge this in.

**Comment by Duncan-Henry** [comment]
Awesome. Is there someone already working on supporting pets? I could probably try to take a look at that.

**Comment by CarlosFdez** [comment]
Currently no one. My immediate thought is to make the familiar text in the top right into a dropdown where the contents are right aligned.

**Comment by CarlosFdez** [comment]
`npm run lint --fix` to fix issues automatically, `npm run test` to see if github will accept it.

File: src/module/actor/familiar/document.ts [hunk 1]
```diff
@@ -141,6 +141,10 @@ class FamiliarPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e
 
         const { level, master, masterAttributeModifier, system } = this;
         const { attributes, traits } = system;
+        const attributeModifier =
+            masterAttributeModifier > 2
+                ? new ModifierPF2e(`PF2E.MasterAbility.${system.master.ability}`, masterAttributeModifier, "untyped")
+                : new ModifierPF2e(`PF2E.Actor.Familiar.MinimumAbilityModifier`, 3, "untyped");
```
Comment (CarlosFdez) [inline src/module/actor/familiar/document.ts:147]:
```suggestion
                : new ModifierPF2e(`PF2E.Actor.Familiar.MinimumAttributeModifier`, 3, "untyped");
```

File: static/lang/en.json [hunk 2]
```diff
@@ -463,6 +463,7 @@
             "Dead": "Dead",
             "Familiar": {
                 "Blurb": "Familiar of {master}",
+                "MinimumAbilityModifier": "Minimum Ability Modifier",
```
Comment (CarlosFdez) [inline static/lang/en.json:466]:
```suggestion
                "MinimumAttributeModifier": "Minimum Attribute Modifier",
```

**Pull request merged**